Improve "upload" function

ViorelSuman opened the pull request:
Update "upload" function in order to be able to handle a specific
"upload" context (like upload a variable value, a byte array, etc.).

nxpfrankli commented on libuuu/fastboot.cpp:
```diff
@@ -266,8 +269,11 @@ int FBUpload::run(CmdCtx* ctx)
-	if (fb.Transport(cmd, nullptr, buff.size(), &buff))
+	if (fb.Transport(cmd, nullptr, buff.size(), &buff)) {
+		string err = "Error on transport !";
+		set_last_err_string(err);
```
All error code setting at leaf node.  For here,  "error on transport" should be set in fb.transport() function, where actuall return -1.

ViorelSuman replied:
Done.

nxpfrankli reviewed (changes requested):
I have not seen uboot's upload implement, which supportted additional parameter?

If it is also added by you, do you have uboot's side patch?  Any defination about this additional parameter

ViorelSuman commented:
> I have not seen uboot's upload implement, which supportted additional parameter?
> 
> If it is also added by you, do you have uboot's side patch? Any defination about this additional parameter

@nxpfrankli , I've added you as reviewer in the pull request I've made in our internal tree. On "uuu" side the optional "-v <some variable name>" just defines the non-default context in which the "upload" command is executed, the default behavior is still in place if "-v ..." option is not passed.
